lib/service/cfg.go

```diff
@@ -477,6 +481,43 @@ func (c ProxyConfig) KubeAddr() (string, error) {
 	return u.String(), nil
 }
 
+// publicPeerAddr attempts to returns the public address the proxy advertises
+// for proxy peering clients if available. It falls back to PeerAddr othewise.
+func (c ProxyConfig) publicPeerAddr() (*utils.NetAddr, error) {
+	var addr *utils.NetAddr
+	if len(c.PeerPublicAddrs) > 0 {
```

Comment:
Is there a reason for having multiple addresses if we always seem to use the first?

Reply:
If we don't use multiple addresses, then let's reflect that in the field declaration.

Let's make sure that, if we copy something, it's for a reason.